owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/property/service/PropertyServiceImpl.java

```diff
@@ -0,0 +1,345 @@
+import fr.openwide.core.commons.util.functional.converter.StringBigDecimalConverter;
+import fr.openwide.core.commons.util.functional.converter.StringBooleanConverter;
+import fr.openwide.core.commons.util.functional.converter.StringDateConverter;
+import fr.openwide.core.commons.util.functional.converter.StringDateTimeConverter;
+import fr.openwide.core.commons.util.functional.converter.StringDirectoryFileConverter;
+import fr.openwide.core.commons.util.functional.converter.StringURIConverter;
+import fr.openwide.core.jpa.exception.SecurityServiceException;
+import fr.openwide.core.jpa.exception.ServiceException;
+import fr.openwide.core.jpa.more.business.parameter.model.Parameter;
+import fr.openwide.core.jpa.more.business.parameter.service.IAbstractParameterService;
+import fr.openwide.core.jpa.more.business.property.dao.IImmutablePropertyDao;
+import fr.openwide.core.jpa.more.business.property.dao.IMutablePropertyDao;
+import fr.openwide.core.jpa.more.business.property.model.ImmutablePropertyId;
+import fr.openwide.core.jpa.more.business.property.model.ImmutablePropertyRegistryKey;
+import fr.openwide.core.jpa.more.business.property.model.MutablePropertyId;
+import fr.openwide.core.jpa.more.business.property.model.PropertyId;
+import fr.openwide.core.jpa.more.business.property.model.PropertyRegistryKey;
+import fr.openwide.core.jpa.more.config.spring.event.PropertyServiceInitEvent;
+import fr.openwide.core.spring.config.CoreConfigurer;
+
+/**
+ * Use this service to retrieve application properties instead of using {@link CoreConfigurer} or {@link IAbstractParameterService}.
+ * It handles both mutable and immutable properties ; immutable properties are retrieved from properties resources files
+ * ({@link IImmutablePropertyDao}) and mutable properties are related to {@link Parameter} ({@link IMutablePropertyDao}).
+ * @see {@link IPropertyRegistry} to register application properties.
+ */
+@Service("propertyService")
+public class PropertyServiceImpl implements IConfigurablePropertyService, ApplicationEventPublisherAware {
+
+	private final Map<PropertyRegistryKey<?>, Pair<? extends Converter<String, ?>, ? extends Supplier<?>>> propertyInformationMap = Maps.newHashMap();
+
+	@Autowired
+	private IMutablePropertyDao mutablePropertyDao;
+	
+	@Autowired
+	private IImmutablePropertyDao immutablePropertyDao;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private TransactionTemplate readOnlyTransactionTemplate;
+
+	private TransactionTemplate writeTransactionTemplate;
+
+	@PostConstruct
+	public void init() {
+		applicationEventPublisher.publishEvent(new PropertyServiceInitEvent(this));
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	@Override
+	public <T> void register(PropertyRegistryKey<T> propertyId, Converter<String, ? extends T> converter) {
```

Comment:
Should be `Converter<String, T>`, not `Converter<String, ? extends T>`, because a converter is two-ways. You don't want it to fail when you ask it to convert a object of type T exactly to a String.